recipes/MEDIA/media_prepare.py

```diff
@@ -0,0 +1,1209 @@
+
+    Arguments
+    ---------
+    data_folder: str
+        Path where folders S0272 and E0024 are stored.
+    save_folder: str
+        Path where the csvs and preprocessed wavs will be stored.
+    skip_wav: bool, optional
+        Skip the wav files storing if already done before.
+    method: str, optional
+        Used only for 'slu' task.
+        Either 'full' or 'relax'.
+        'full' Keep specifiers in concepts.
+        'relax' Remove specifiers from concepts.
+    task: str, optional
+        Either 'slu' or 'asr'.
+        'slu' Parse SLU data.
+        'asr' Parse ASR data.
+    skip_prep: bool, optional
+        If True, skip data preparation.
+    """
+
+    if skip_prep:
+        return
+
+    if not os.path.exists(save_folder):
+        os.makedirs(save_folder)
+    if not os.path.exists(save_folder + "/wav"):
+        os.makedirs(save_folder + "/wav")
+    if not os.path.exists(save_folder + "/csv"):
+        os.makedirs(save_folder + "/csv")
+
+    if skip(
+        save_folder + "/csv/train.csv",
+        save_folder + "/csv/dev.csv",
+        save_folder + "/csv/test.csv",
+    ):
+        logger.info("Csv files already exist, skipping data preparation!")
+        return
+
+    if task == "slu":
+        if method == "relax" or method == "full":
+            logger.info("Processing SLU " + method + " Media Dataset")
+        else:
+            raise ValueError("Parameter method must be 'full' or 'relax'")
+    elif task == "asr":
+        logger.info("Processing ASR Media Dataset")
+    else:
+        raise ValueError("Parameter task must be 'asr' or 'slu'")
+
+    xmls = {
+        "media_lot1.xml": "train",
+        "media_lot2.xml": "train",
+        "media_lot3.xml": "train",
+        "media_lot4.xml": "train",
+        "media_testHC.xml": "test",
+        "media_testHC_a_blanc.xml": "dev",
+    }
+
+    wav_paths = glob.glob(data_folder + "/S0272/**/*.wav", recursive=True)
+    channels, filenames = get_channels("./channels.csv")
+    unused_dialogs = get_unused_dialogs(data_folder)
+    concepts_full, concepts_relax = get_concepts_full_relax(
+        "./concepts_full_relax.csv"
+    )
+
+    write_first_row(save_folder)
+
+    # Wavs.
+    if not (skip_wav):
+        logger.info("Processing wavs")
+        for wav_path in tqdm(wav_paths):
+            filename = wav_path.split("/")[-1][:-4]
+            channel = get_channel(filename, channels, filenames)
+            split_audio_channels(wav_path, filename, channel, save_folder)
+
+    # Train, Dev, Test.
+    for xml in xmls:
+        logger.info(
+            "Processing file "
+            + str(list(xmls.keys()).index(xml) + 1)
+            + "/"
+            + str(len(xmls))
+        )
+        root = get_root(
+            data_folder + "/E0024/MEDIA1FR_00/MEDIA1FR/DATA/" + xml, 0,
+        )
+        parse(
+            root, channels, filenames, save_folder, method, task, xmls[xml],
+        )
+
+    # Test2.
```

Comment:
the test2 should be optional. A user might want to only preprocess the "official" test set